_config.yml

```diff
@@ -76,6 +76,28 @@ sphinx:
     # application/vnd.holoviews_load.v0+json, application/vnd.holoviews_exec.v0+json
     suppress_warnings: ["mystnb.unknown_mime_type", "misc.highlighting_failure"]
     codeautolink_concat_default: True
+    codeautolink_warn_on_missing_inventory: True
+    codeautolink_warn_on_failed_resolve: True
+    codeautolink_inventory_map: {
+        # unfortunately mapping the top level doesn't work, need all methods
+        #"xarray.core.dataarray.DataArray": "xarray.DataArray",
+        "xarray.core.accessor_dt.DatetimeAccessor": "xarray.DataArray.dt",
+        "xarray.core.dataarray.DataArray.coarsen": "xarray.DataArray.coarsen",
+        "xarray.core.dataarray.DataArray.groupby": "xarray.DataArray.groupby",
+        "xarray.core.dataarray.DataArray.groupby_bins": "xarray.DataArray.groupby_bins",
```

Comment:
@felix-hilden, I realize this issue (https://github.com/felix-hilden/sphinx-codeautolink/issues/131#issuecomment-1407159186 ) has been closed for a while, but curious based on your comment if it does seems straightforward to implement a simplification of ("xarray.core.dataarray.DataArray": "xarray.DataArray") to avoid having to enumerate every method?

Reply:
Would setting `__module__` fix this? (https://github.com/pydata/xarray/issues/4279)

Reply:
Based on the above issue, if I add
```python
import xarray 
xarray.DataArray.__module__ = "xarray"
```
to `conf.py` before building I'm getting `Handler <bound method SphinxCodeAutoLink.create_references of <sphinx_codeautolink.extension.SphinxCodeAutoLink object at 0x145730710>> for event 'env-updated' threw an exception (exception: maximum recursion depth exceeded)` ...

Reply:
Can you try adding it to xarrays init.py?

Reply:
Hi! Yeah I'm not at all opposed to having the feature that was discussed in the issue, I've just been out of the game for a while focusing on other things and I have personally no need for that. But I'm very open to contributions 👌 

Personally I've opted for changing module attributes [like so](https://github.com/felix-hilden/tekore/blob/98d40481d036c4b35b31579d83dbfe35ac44600d/src/tekore/model.py#L214) and it's worked well for me.

Reply:
actually it does look like chaning either `conf.py` or `xarray/__init__.py` works, I think the recursion error was coming from some issue with my local settings/cache...

Reply:
kind of a pain to modify conf.py via jupyter book (https://github.com/executablebooks/jupyter-book/issues/858,
https://github.com/executablebooks/jupyter-book/issues/1673#issuecomment-1199090107) but that is probably the way to go...

Reply:
Let's do it upstream? I think we'd like this in Xarray too.